client/src/containers/transactions/components/form/index.tsx

```diff
@@ -0,0 +1,364 @@
+/* eslint @typescript-eslint/indent: 0,
+   react/jsx-curly-newline: 0 */ // Linter and prettier conflicts, should be fixed.
+import React from 'react';
+import {
+  TextField,
+  Button,
+  Container,
+  Select,
+  MenuItem,
+  InputLabel,
+  FormControl,
+  FormHelperText,
+} from '@material-ui/core';
+import { Autocomplete } from '@material-ui/lab';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import DateFnsUtils from '@date-io/date-fns';
+import { State } from 'store';
+import {
+  KeyboardDatePicker,
+  MuiPickersUtilsProvider,
+} from '@material-ui/pickers';
+import {
+  TransactionCategories,
+  Transaction,
+  createTransaction,
+  TransactionFormErrors,
+  setTransactionFormErrors,
+  clearTransactionFormErrors,
+  editTransaction,
+} from 'store/modules/transactions';
+import Routes from 'utils/routes';
+import {
+  AccountSelectOption,
+  getAccountSelectOptions,
+} from 'store/modules/accounts';
+import style from './style.module.scss';
+
+interface IState {
+  amount: number | '';
+  date: Date;
+  category: TransactionCategories | '';
+  description: string;
+  accountOption: AccountSelectOption | null;
+  accountSelectInput: string;
+}
```

Comment:
Kodėl neišsikėlus į types failą? Arba į store types pavadinant interfacą IAccountState?

Reply:
Paprastai komponento vidinis state tam pačiam faile būna declarinamas - čia formos komponento vidinis state

Reply:
Got it